File: C++/2846.cpp
#include <bits/stdc++.h>

using namespace std;

#define MAXSIZE 100100

int fibonnaci[MAXSIZE];
int fibonot[MAXSIZE];

void fib() {
    fibonnaci[0] = 0;
    fibonnaci[1] = 1;

    for (int i = 2; i < MAXSIZE; i++) {
        fibonnaci[i] = fibonnaci[i - 1] + fibonnaci[i - 2];
    }

    int k = 1;
    int j = 2;

    for (int i = 1; i <= MAXSIZE; i++) {
        if (fibonnaci[j] != i) {
            fibonot[k] = i;
            k++;
        } else {
            j++;
        }
    }
}

int main() {
    fib();
    int n;
    cin >> n;
    cout << fibonot[n] << endl;
    return 0;
}

File: Go/2846.go
package main

import "fmt"

const MAXSIZE = 100100

var fibonacci []int
var fibonot []int

func main() {
	var n int
	fib()

	fmt.Scanf("%d", &n)
	fmt.Printf("%d\n", fibonot[n])
}

func fib() {
	fibonacci = make([]int, MAXSIZE)
	fibonot = make([]int, MAXSIZE)

	fibonacci[0] = 0
	fibonacci[1] = 1

	for i := 2; i < MAXSIZE; i++ {
		fibonacci[i] = fibonacci[i-1] + fibonacci[i-2]
	}

	k := 1
	j := 2

	for i := 1; i <= MAXSIZE; i++ {
		if fibonacci[j] != i {
			fibonot[k] = i
			k++
		} else {
			j++
		}
	}
}
